### DISCUSSION TITLE
Issue with using js window interface in in-app browser on iOS using cap Browser plugin

### DISCUSSION THREAD

**Discussion by jellis206** [ionic-team/capacitor#7283] [Q&A] [answered]
At work, we are integrating with a third party, we needed to redirect them to their website so we decided to use an in app browser session using the Capacitor Browser plugin (the user is just filling out some forms on their website). For legal reasons the user needs to be able to print out some of the terms and conditions. A coworker and I did some exploring and saw that in the browser window (opened with the Browser plugin) `window.print()` works on android, however `window.print()` does not work on iOS (pretty easily found just by opening dev tools for the respective os's). After looking into it it looks like the issue is related to SFSafariViewController. We tested a few other window functions and many of them are just undefined, while a few of them work as expected. I was wondering if anyone in the community has ever dealt with this / if anyone knows how to make that functionality available. I wanted to ask before opening an issue or trying to extend the plugin (if that is even possible).

tldr -> anyone know how to get `window.print()` or other js window methods to work in an in app browser session using cap browser plugin on iOS?

**Comment by jellis206** [comment]
It might be worth mentioning that if the Browser Plugin were refactored to use WKWebView instead of SFSafariViewController this would likely be solved. From what I read in the SFSafariViewController docs is that WKWebView should be used, "If you need to customize the controls of the web interface, or you want to interact with content in that interface..." I would do this refactor myself if I had any experience with native iOS dev and swift. Maybe it is time to go learn...

**Reply by jellis206** [reply]
Even this might not be true, we tried @capgo/inappbrowser which uses the WKWebView and that didn't make things any better.

**Comment by jellis206** [comment, marked as answer]
Ionic answered me through a different channel:

So this is a known bug with SFSafariViewController. The user would need to use the share dialog to print. Alternatively, you can try using the [cordova InAppBrowser plugin](https://github.com/apache/cordova-plugin-inappbrowser/tree/master), which uses WKWebView instead, and see if that has a working print API.
